mpas_analysis/shared/plot/plotting.py

```diff
@@ -44,35 +47,48 @@ def timeseries_analysis_plot(config, dsvalues, N, title, xlabel, ylabel, fileout
 
     Author: Xylar Asay-Davis
 
-    Last Modified: 02/02/2017
+    Last Modified: 02/11/2017
     """
-    plt.figure(figsize=figsize, dpi=dpi)
+    fig = plt.figure(figsize=figsize, dpi=dpi)
 
     for dsIndex in range(len(dsvalues)):
         dsvalue = dsvalues[dsIndex]
         if dsvalue is None:
             continue
         mean = pd.Series.rolling(dsvalue.to_pandas(), N, center=True).mean()
-        mean.plot(style=lineStyles[dsIndex], lw=lineWidths[dsIndex])
+        mean = xr.DataArray.from_series(mean)
+        plt.plot_date(mean['Time'], mean,
+                      fmt=lineStyles[dsIndex],
+                      xdate=True,
+                      ydate=False,
+                      linewidth=lineWidths[dsIndex])
+
+    ax = plt.gca()
+    # TODO: it would be good to change labels to include months if the plot is
+    # shorter than a couple of years
+    ax.xaxis.set_major_locator(mdates.YearLocator())
+    ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y'))
+    ax.xaxis.set_minor_locator(mdates.MonthLocator())
+    fig.autofmt_xdate()
 
     if titleFontSize is None:
         titleFontSize = config.get('plot', 'titleFontSize')
 
-    axis_font = {'size':config.get('plot', 'axisFontSize')}
+    axis_font = {'size': config.get('plot', 'axisFontSize')}
     title_font = {'size': titleFontSize,
-                  'color':config.get('plot', 'titleFontColor'),
-                  'weight':config.get('plot', 'titleFontWeight')}
-    if (title != None):
+                  'color': config.get('plot', 'titleFontColor'),
+                  'weight': config.get('plot', 'titleFontWeight')}
+    if title is not None:
```

Comment:
Regarding "is not None" vs "!= None", I'm quite certain we never reached the opposite conclusion because this is a standard I have known about for quite awhile.  If you can point me to the discussion you're referring to, that would be helpful.